scripts/data_utils.py

```diff
@@ -436,21 +436,30 @@ def merge_chunks_serially(chunked_content_list: List[str], num_tokens: int) -> G
         yield current_chunk, total_size
 
 
-def get_embedding(text, azure_credential, embedding_endpoint):
+def get_embedding(text, embedding_endpoint, embedding_api_key):
     try:
         endpoint_parts = embedding_endpoint.split("/openai/deployments/")
         base_url = endpoint_parts[0]
         deployment_id = endpoint_parts[1].split("/embeddings")[0]
 
         openai.api_version = '2023-05-15'
         openai.api_base = base_url
-        openai.api_key = azure_credential.get_token("https://cognitiveservices.azure.com/.default").token
-        openai.api_type = "azure_ad"
-
-        embeddings = openai.Embedding.create(deployment_id=deployment_id, input=text)
-        return embeddings['data'][0]["embedding"]
+        openai.api_key = embedding_api_key
+        openai.api_type = "azure"
 
+        # Retry while embedding requests get throttled
+        retry_count = 0
+        # while retry_count < RETRY_COUNT:
+        while True:
+            try:
+                embeddings = openai.Embedding.create(deployment_id=deployment_id, input=text)
+                return embeddings['data'][0]["embedding"]
+            except openai.error.RateLimitError as throttled:
+                print(f'Retry attempt {retry_count}: Embedding API throttling, waiting for 4 second before retry...')

```

Comment:
We use the tenacity library when we do need backoff, as it will do exponential backoff. But as I said, we rarely need to backoff now that we have the batching change.